numpy/core/src/private/get_attr_string.h

```diff
@@ -1,37 +1,55 @@
 #ifndef __GET_ATTR_STRING_H
 #define __GET_ATTR_STRING_H
 
-static NPY_INLINE int
-_is_basic_python_type(PyObject * obj)
+static NPY_INLINE npy_bool
+_is_basic_python_type(PyTypeObject *tp)
 {
-    if (obj == Py_None ||
-            PyBool_Check(obj) ||
-            /* Basic number types */
+    PyTypeObject const * const known_types[] = {
```

Comment:
I'll wait for the tests to pass before pushing that, to see if I missed anything that my local build din't pick up. Any other comments?